src/pycolmap/scene/imu.cc

```diff
@@ -0,0 +1,123 @@
+        std::stringstream ss;
+        ss << "ImuMeasurement("
+           << "t=" << m.timestamp << ", "
+           << "acc=[" << m.linear_acceleration.format(vec_fmt) << "], "
+           << "gyro=[" << m.angular_velocity.format(vec_fmt) << "])";
+        return ss.str();
+      });
+  MakeDataclass(PyImuMeasurement);
+
+  py::class_<ImuMeasurements>(m, "ImuMeasurements")
+      .def(py::init<>())
+      .def(py::init<const std::vector<ImuMeasurement>&>())
+      .def(py::init<const ImuMeasurements&>())
+      .def("insert",
+           py::overload_cast<const ImuMeasurement&>(&ImuMeasurements::insert))
+      .def("insert",
+           py::overload_cast<const std::vector<ImuMeasurement>&>(
+               &ImuMeasurements::insert))
+      .def("insert",
+           py::overload_cast<const ImuMeasurements&>(&ImuMeasurements::insert))
+      .def("remove", &ImuMeasurements::remove)
+      .def("front", &ImuMeasurements::front)
+      .def("back", &ImuMeasurements::back)
+      .def("empty", &ImuMeasurements::empty)
+      .def("clear", &ImuMeasurements::clear)
+      .def("__len__", &ImuMeasurements::size)
+      .def("__getitem__",
+           [](const ImuMeasurements& ms, size_t idx) { return ms[idx]; })
+      .def("get_measurements_contain_edge",
+           &ImuMeasurements::GetMeasurementsContainEdge)
+      .def_property_readonly("data", &ImuMeasurements::Data)
+      .def("__repr__", [](const ImuMeasurements& ms) {
+        std::stringstream ss;
+        ss << "ImuMeasurements("
+           << "n=" << ms.size() << ", "
+           << "tmin=" << ms.front().timestamp << ", "
+           << "tmax=" << ms.back().timestamp << ")";
+        return ss.str();
+      });
+
+  py::class_<Imu>(m, "Imu")
+      .def(py::init<>())
+      .def_readwrite("imu_id", &Imu::imu_id)
+      .def_readwrite("camera_id", &Imu::camera_id)
+      .def_readwrite("imu_from_cam", &Imu::imu_from_cam)
+      .def("__repr__", [](const Imu& s) {
+        std::stringstream ss;
+        ss << "Imu("
+           << "imu_id=" << s.imu_id << ", "
+           << "camera_id=" << s.camera_id << ")";
+        return ss.str();
+      });
+
+  py::class_<ImuState>(m, "ImuState")
+      .def(py::init<>())
+      .def("set_velocity", &ImuState::SetVelocity)
+      .def("set_acc_bias", &ImuState::SetAccBias)
+      .def("set_gyro_bias", &ImuState::SetGyroBias)
+      .def_property(
+          "data",
+          py::overload_cast<>(&ImuState::Data),
+          [](ImuState& self, const Eigen::Matrix<double, 9, 1>& data) {
+            self.Data() = data;
+          })
+      .def_property_readonly("velocity", &ImuState::Velocity)
+      .def_property_readonly("velocity_ptr", &ImuState::VelocityPtr)
+      .def_property_readonly("acc_bias", &ImuState::AccBias)
+      .def_property_readonly("acc_bias_ptr", &ImuState::AccBiasPtr)
+      .def_property_readonly("gyro_bias", &ImuState::GyroBias)
+      .def_property_readonly("gyro_bias_ptr", &ImuState::GyroBiasPtr)
```

Comment:
You can make the properties writable and remove the `set_*` methods.

Reply:
The three properties here are continuous data under Eigen::Vector9d. To be able to do ``set_*`` we require the input to be Eigen::Vector3d to avoid invalid access of data.